src/actions/value.js

```diff
@@ -38,7 +38,16 @@ export function setValue(
 			componentTypes.reactiveMap,
 		].includes(props[component] && props[component].componentType);
 		const previousValue = selectedValues[component] && selectedValues[component].value;
-		if (!isEqual(previousValue, value) && props[component] && !isResultComponent) {
+		// Incase of searchbox .meta.imageValue can be modified although .value is not modified.
+		let isImageValueEqual = true;
+		// SelectedFilters may not have componentType but they need to clear the value
```

Comment:
_// SelectedFilters may not have componentType but they need to clear the value_

Can you point to an example when this can happen?

Reply:
Evidently, it is always the case that SelectedFilters don't have a componentType. You can see that in the [componentTypes](https://github.com/appbaseio/reactivecore/blob/master/src/utils/constants.js#L1) constant.
Else the line is a guard to not allow just any other component having a similar named property `imageValue` to change the `imageValue`.